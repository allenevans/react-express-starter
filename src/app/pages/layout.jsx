/* eslint-disable import/no-named-as-default,no-unused-vars */
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import './layout.scss';
import Footer from '../components/footer/footer';
import Header from '../components/site-header/site-header';
import { windowResize } from '../redux/actions/ui/ui-actions';

export class Layout extends Component {
  constructor(props) {
    super(props);

    this.resizeHandler = () => {
      this.props.actions.windowResize();
    };
  }

  componentDidMount() {
    window.addEventListener('resize', this.resizeHandler, false);
  }

  componentDidUpdate() {
    if (this.props.loading) {
      document.body.classList.add('network-loading');
    } else {
      document.body.classList.remove('network-loading');
    }
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.resizeHandler, false);
  }

  render() {
    return (
      <div className="layout">
        <Header />
        <main className="main-content">
          <div className={this.props.className}>
            {this.props.children}
          </div>
        </main>

        {!this.props.hideFooter ? <Footer /> : null}
      </div>
    );
  }
}

Layout.propTypes = {
  actions: PropTypes.object,
  children: PropTypes.any,
  className: PropTypes.string,
  hideFooter: PropTypes.bool,
  loading: PropTypes.bool,
  windowHeight: PropTypes.number,
};

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators({
    windowResize,
  }, dispatch),
});

const mapStateToProps = state => ({
  windowHeight: state.ui.windowHeight,
});

export default connect(mapStateToProps, mapDispatchToProps)(Layout);
