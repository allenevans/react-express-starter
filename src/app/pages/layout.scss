.layout {
  display: flex;
  flex-direction: column;

  &.loading {
    overflow: hidden;
  }
}
