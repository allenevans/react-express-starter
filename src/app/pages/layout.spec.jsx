import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import { shallow } from 'enzyme';
import { Layout } from './layout';

const renderer = new ReactShallowRenderer();

describe('Layout component', () => {
  it('should render the component', () => {
    const tree = renderer.render(<Layout />);

    expect(tree).toMatchSnapshot();
  });

  it('should render children passed into the Layout component', () => {
    const enzymeWrapper = shallow(<Layout>
        <h1>HEADING</h1>
        <p>CONTENT</p>
      </Layout>);

    expect(enzymeWrapper.find('h1').text()).toBe('HEADING');
    expect(enzymeWrapper.find('p').text()).toBe('CONTENT');
  });
});
