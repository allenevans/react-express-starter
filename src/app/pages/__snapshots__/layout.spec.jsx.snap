// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component should render the component 1`] = `
<div
  className="layout"
>
  <Connect(SiteHeader) />
  <main
    className="main-content"
  >
    <div
      className=""
    />
  </main>
  <SiteFooter />
</div>
`;
